10.1/root/usr/bin/run-mysqld-slave

```diff
@@ -48,8 +48,14 @@ mysql $mysql_flags <<EOSQL
   START SLAVE;
 EOSQL
 
-log_info 'Sourcing post-init.sh ...'
-[ -f ${CONTAINER_SCRIPTS_PATH}/post-init.sh ] && source ${CONTAINER_SCRIPTS_PATH}/post-init.sh
+if [ -f ${CONTAINER_SCRIPTS_PATH}/post-init.sh ]; then
+  log_info 'Sourcing post-init.sh ...'
+  source ${CONTAINER_SCRIPTS_PATH}/post-init.sh
+fi
+if [ -f ${CONTAINER_SCRIPTS_PATH}/check-upgrade.sh ]; then
+  log_info 'Sourcing check-upgrade.sh ...'
+  source ${CONTAINER_SCRIPTS_PATH}/check-upgrade.sh
+fi
```

Comment:
Agreed. The user could run some script as part of post-init that could fail when the data is not upgraded.